pkg/ui/react-app/src/components/ListTree.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { InputProps, Collapse, ListGroupItem, ListGroup } from 'reactstrap';
+
+export interface NodeTree {
+  name: string;
+  children?: NodeTree[];
+}
+
+interface NodeProps extends InputProps {
+  node: NodeTree | null;
+}
+
+const ListTree: React.FC<NodeProps> = ({ id, node }) => {
+  type mapping = {
+    [key: string]: boolean;
+  };
+
+  const [state, setState] = useState<mapping>({});
+  const toggle = (e: React.MouseEvent<HTMLDivElement>) => {
+    const el = e.target as HTMLDivElement;
+    const id = el.getAttribute('id');
+    if (id) {
+      setState({ ...state, [id]: !state[id] });
+    }
+  };
+
+  const mapper = (nodes: NodeTree[], parentId?: any, lvl?: any) => {
+    return nodes.map((node: NodeTree, index: number) => {
+      const id = `${index}-${parentId ? parentId : 'top'}`.replace(/[^a-zA-Z0-9-_]/g, '');
```

Comment:
Why are we constructing the id this way?

Reply:
I don't know much, I found it online and changed it a little.